.github/workflows/ci-build.yml

```diff
@@ -20,34 +15,34 @@ jobs:
       - uses: actions/setup-java@v2
         with:
           distribution: 'zulu'
-          java-version: ${{ matrix.java }}
+          java-version: 11
 
       - name: Cache local Maven repository
         uses: actions/cache@v2
         with:
           path: ~/.m2/repository
-          key: ${{ runner.os }}-java-${{ matrix.java }}-maven-${{ hashFiles('**/pom.xml') }}
+          key: ${{ runner.os }}-java-11-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
-            ${{ runner.os }}-java-${{ matrix.java }}-maven-
+            ${{ runner.os }}-java-11-maven-
 
       - name: Cache SonarQube packages
         uses: actions/cache@v2
         with:
           path: ~/.sonar/cache
-          key: ${{ runner.os }}-java-${{ matrix.java }}-sonar
-          restore-keys: ${{ runner.os }}-java-${{ matrix.java }}-sonar
+          key: ${{ runner.os }}-java-11-sonar
+          restore-keys: ${{ runner.os }}-java-11-sonar
 
       - name: Run tests and build with Maven
         run: mvn --batch-mode --update-snapshots clean verify -DtrimStackTrace=false
       
-      - name: Publish Test Report for Java ${{ matrix.java }}
+      - name: Publish Test Report for Java 11
         uses: scacap/action-surefire-report@v1
         if:  ${{ always() }}
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Sonar analysis
-        if: ${{ env.DEFAULT_JAVA == matrix.java && env.SONAR_TOKEN != null }}
+        if: ${{ env.SONAR_TOKEN != null }}
```

Comment:
I think it's better if it fails In case there is no token. Then we know it's missing